.github/workflows/check.yml

```diff
@@ -13,34 +13,98 @@ jobs:
-        run: mysql -h 127.0.0.1 --port 3306 -uroot -proot -e "show databases;"
-      - name: Create database
-        run: mysql -h 127.0.0.1 --port 3306 -uroot -proot -e "create database ci_test;"
-      - name: Build with testing
-        run: ./gradlew --stacktrace :embulk-output-mysql:check
+    - uses: actions/checkout@v4
+    - name: Set up JDK 8
+      uses: actions/setup-java@v4
+      with:
+        java-version: 8
+        distribution: 'zulu'
+        cache: "gradle"
+    - name: Connect
+      run: mysql -h 127.0.0.1 --port 3306 -uroot -proot -e "show databases;"
+    - name: Create database
+      run: mysql -h 127.0.0.1 --port 3306 -uroot -proot -e "create database ci_test;"
+    - name: Build with testing
+      run: ./gradlew --stacktrace :embulk-output-mysql:check
+      env:
+        _JAVA_OPTIONS: "-Xmx2048m -Xms512m"
+        EMBULK_OUTPUT_MYSQL_TEST_CONFIG: "${{ github.workspace }}/ci/mysql.yml"
+    - uses: actions/upload-artifact@v4
+      if: always()
+      with:
+        name: mysql5_7
+        path: embulk-output-mysql/build/reports/tests/test
+  mysql8_3:
+    runs-on: ubuntu-latest
+    # push: always run.
+    # pull_request: run only when the PR is submitted from a forked repository, not within this repository.
+    if: github.event_name == 'push' || github.event.pull_request.head.repo.full_name != github.repository
+    strategy:
+      fail-fast: false
+    services:
+      mysql:
+        # Due to MySQL 8.4 disabled mysql_native_password by default,
+        # Connector/J 5.x can't connect to database.
+        # So, Use MySQL 8.3.
+        image: mysql:8.3
+        options: --health-cmd "mysqladmin ping -h localhost" --health-interval 20s --health-timeout 10s --health-retries 10
+        ports:
+        - "3306:3306"
         env:
-          _JAVA_OPTIONS: "-Xmx2048m -Xms512m"
-          EMBULK_OUTPUT_MYSQL_TEST_CONFIG: "${{ github.workspace }}/ci/mysql.yml"
-      - uses: actions/upload-artifact@v3
-        if: always()
-        with:
-          name: mysql
-          path: embulk-output-mysql/build/reports/tests/test
-  postgresql:
+          MYSQL_ROOT_PASSWORD: root
+          MYSQL_USER: ci
+          MYSQL_PASSWORD: password
+    steps:
+    - uses: actions/checkout@v4
+    - name: Set up JDK 8
+      uses: actions/setup-java@v4
+      with:
+        java-version: 8
+        distribution: 'zulu'
+        cache: "gradle"
+    - name: Connect
+      run: mysql -h 127.0.0.1 --port 3306 -uroot -proot -e "show databases;"
+    - name: show version
+      run: mysql -h 127.0.0.1 --port 3306 -uroot -proot -e "select version();"
+    - name: Create database
+      run: mysql -h 127.0.0.1 --port 3306 -uroot -proot -e "create database ci_test;"
+    #
+    # MySQL 8 uses caching_sha2_password mechanism by default.
+    # Connector/J 5.x doesn't support it.
+    #
+    # This part change password mechanism to mysql_native_password.
+    # Remove the following part after update Connector/J
+    #
```

Comment:
```suggestion
    # Workaround to change MySQL's password mechanism to `mysql_native_password`
    # from `caching_sha2_password` that is the default in MySQL 8 because
    # Connector/J 5.x does not support `caching_sha2_password`.
    # TODO: Start testing with `caching_sha2_password` with Connector/J 8.x.
```